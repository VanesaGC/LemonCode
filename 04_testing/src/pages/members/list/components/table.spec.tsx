import * as React from "react";
import { shallow } from "enzyme";
import { Table } from "./table";

describe("pagesmemberslistcomponents (table) tests", () => {
  it("should render as expected when passing required properties", () => {
    // Arrange
    const props = {
      members: [
        {
          id: 1,
          name: "Mar√≠a",
          avatarUrl: "maria.png"
        },
        {
          id: 2,
          name: "Pedro",
          avatarUrl: "pedro.png"
        }
      ]
    };
    // Act
    const component = shallow(<Table {...props} />);
    // Assert
    expect(component).toMatchSnapshot();
  });
});
