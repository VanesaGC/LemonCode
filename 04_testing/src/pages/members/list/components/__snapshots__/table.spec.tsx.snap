// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagesmemberslistcomponents (table) tests should render as expected when is invoked 1`] = `
<table
  className="table table-striped"
>
  <Component />
  <Component
    members={
      Array [
        Object {
          "avatarUrl": "maria.png",
          "id": 1,
          "name": "María",
        },
        Object {
          "avatarUrl": "pedro.png",
          "id": 2,
          "name": "Pedro",
        },
      ]
    }
  />
</table>
`;

exports[`pagesmemberslistcomponents (table) tests should render as expected when passing required properties 1`] = `
<table
  className="table table-striped"
>
  <Component />
  <Component
    members={
      Array [
        Object {
          "avatarUrl": "maria.png",
          "id": 1,
          "name": "María",
        },
        Object {
          "avatarUrl": "pedro.png",
          "id": 2,
          "name": "Pedro",
        },
      ]
    }
  />
</table>
`;
